src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -136,16 +136,16 @@ protected void setPrivateFieldValue(String fieldName, Object newValue) {
         return getPrivateFieldValue("additionalCommands", Map.class);
     }
 
-    protected CommandCodec<HttpRequest> getCommandCodec() {
```

Comment:
I'd rather keep them protected if possible

Reply:
at least setters

Reply:
How could that be achieved here?

Reply:
Oh, I mean. I figured that as long as the setters were made public, you might as well get access to the getters. It's not a big secret.

Reply:
this is not a secret, but rather protection from shutting your leg. Setting an incorrect codec in runtime will break the driver completely. 
Also I expect public methods to be properly documented

Reply:
these can also be made package-private

Reply:
The executor and the AppiumDriver are in different packages though :(